pkg/diagnose/runner.go

```diff
@@ -49,3 +49,39 @@ func RunAll(w io.Writer) error {
 
 	return nil
 }
+
+// Run runs a specific list of connectivity checks, output it in writer
+func Run(w io.Writer, checks []string) error {
```

Comment:
Why don't you reuse the above `RunAll` function? A lot of duplicated code between the two 🤔

Reply:
that's right some code has been duplicated but is it useful to run all checks for DCA? especially cloud providers metadata checks..
I suggest I refactor `RunAll` and `Run` and create helper function(s) that could be used by both `RunAll` and `Run`
wdyt?

Reply:
@ahmed-mez did you check the list of which diagnosis would be executed? Some of them are behind build flags that might be excluded in the cluster agent binary.

Reply:
thanks for mentioning this, currently (in this PR) we only run diagnose on apiserver connectivity which is included

Reply:
I guess we can keep the `RunAll` function then?

Reply:
makes sense 👍